### ISSUE TITLE
[roq-api] MbP and MbO updates must include price/quantity decimals

### ISSUE THREAD

**Issue by thraneh** [roq-trading/roq-issues#119]
Publishing decimals has become necessary as it became more evident that reference data can never be a guaranteed solution to converting floating point values to some internal integer based representation.

This is particular the case where prices must be stored in an ordered data structure taking into account the rounding errors naturally occurring when working with floating point numbers.

The particular problem discussed in #114 has been the trigger for this issue.

Avoiding downstream confusion, it's now integral to MbP and MbO updates to include the latest decimal digits required to represent floating point numbers using integers.

There are some restrictions on this

* `ReferenceData` *may* be used to initialise the number of decimal digits (but it is no longer a requirement).
* After a snapshot, the number of decimal digits can only increase (because the opposite would mean a likely loss of already cached information).

In short, avoiding fuzzy logic on the client side, requires gateways to publish enough information for the client to (easily) detect state *changes* without resorting to heuristics. This should allow the client to maintain an exact cached copy of the state maintained by the gateways.

**Comment by thraneh** [comment]
This is now implemented as per above.

> Requires extensive testing!

Solved:

* Decimals are now communicated to client
* `ReferenceData` is now only used by server

Future ideas:
* `GatewaySettings` do not have to be communicated to client if max-depth is part of MbP/MbO updates.
